1-raw-data/mailinglists/googlegroups/googlegroups_data.go

```diff
@@ -0,0 +1,429 @@
+
+	var msgId string
+
+	msgUrl, _ := dom.Find("a").Attr("href")
+	if regTopicURL.MatchString(msgUrl) {
+		msgId = path.Base(msgUrl)
+		rawMsgUrl = fmt.Sprintf("https://groups.google.com%s/forum/message/raw?msg=%s/%s/%s", org, groupName, topicId, msgId)
+	}
+	return
+}
+
+// Parse topic ids from dom, get message ids and create raw message url map by year-month filename
+func topicIDToRawMsgUrlMap(org, groupName string, dom *goquery.Document) (rawMsgUrlMap map[string][]string, err error) {
+
+	rawMsgUrlMap = make(map[string][]string)
+
+	regTopicURL, _ := regexp.Compile(fmt.Sprintf("/d/topic/%s", groupName))
+	// Alternate time option [0-9]{1,2}:[0-9]{2}\s(AM|PM)
+	regTime, _ := regexp.Compile("[0-1]{0,1}[0-9]{1,2}:[0-5][0-9] (AM|PM)")
+	regDate, _ := regexp.Compile("[0-9]{1,2}/[0-9]{1,2}/[0-9]{2,4}")
+
+	var fileName, dateToParse, topicID, msgURL, rawMsgURL string
+
+	dom.Find("tr").Each(func(i int, row *goquery.Selection) {
+		row.Find("td").Each(func(i int, cell *goquery.Selection) {
+			topicIdURL, ok := cell.Find("a").Attr("href")
+			if ok {
+				// Get topic id
+				if regTopicURL.MatchString(topicIdURL) {
+					// Capture topic id
+					topicID = path.Base(topicIdURL)
+				}
+			}
+			// Capture date topic posted and convert to year-month text filename for grouping
+			dateClass, _ := cell.Attr("class")
+			if dateClass == "lastPostDate" {
+				matchDate := cell.Text()
+				// If the date is a time then its today otherwise parse the date
+				if regTime.MatchString(matchDate) {
+					dateToParse = time.Now().Format("01/02/06")
+				} else if regDate.MatchString(matchDate) {
+					dateToParse = matchDate
+				}
+				if fileName, err = getFileName(dateToParse); err != nil {
+					err = fmt.Errorf("Defining month year string key for topic lists returned an error: %v", err)
+					return
+				}
+
+				msgURL = fmt.Sprintf("https://groups.google.com/forum/?_escaped_fragment_=topic/%s/%s", groupName, topicID)
+
+				if dom, err = httpDomResponse(msgURL); err != nil {
+					return
+				}
+
+				// Get the message ids from the links associated with each topic id
+				if rawMsgURL, err = getMsgIDsFromDom(org, topicID, groupName, dom); err != nil {
+					return
+				}
+				// Store the urls for the raw message content into a map grouped by year-month
+				rawMsgUrlMap[fileName] = append(rawMsgUrlMap[fileName], rawMsgURL)
+			}
+		})
+	})
+	return
+}
+
+// Worker converting list of topic ids urls to dom objects and converting topic ids into the raw message urls
+func getRawMsgURLWorker(org, groupName string, topicURLJobs <-chan string, results chan<- urlResults) {
+	var (
+		dom                      *goquery.Document
+		topicResults, tmpResults map[string][]string
+		err                      error
+	)
+	topicResults = make(map[string][]string)
+	tmpResults = make(map[string][]string)
+
+	for url := range topicURLJobs {
+		if dom, err = httpDomResponse(url); err != nil {
+			results <- urlResults{err: err}
+			return
+		}
+
```

Comment:
A comment explaining the logic here might be helpful (was this the pagination part?)